src/open_inwoner/configurations/bootstrap/auth.py

```diff
@@ -0,0 +1,482 @@
+from django.conf import settings
+from django.contrib.admin.sites import AdminSite
+from django.contrib.auth.models import Group
+from django.core.exceptions import ValidationError
+from django.test import RequestFactory
+
+from digid_eherkenning.admin import (
+    DigidConfigurationAdmin,
+    EherkenningConfigurationAdmin,
+)
+from digid_eherkenning.models import DigidConfiguration, EherkenningConfiguration
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed
+from mozilla_django_oidc_db.forms import OpenIDConnectConfigForm
+from mozilla_django_oidc_db.models import OpenIDConnectConfig
+from simple_certmanager.models import Certificate
+
+from digid_eherkenning_oidc_generics.admin import (
+    OpenIDConnectDigiDConfigForm,
+    OpenIDConnectEHerkenningConfigForm,
+)
+from digid_eherkenning_oidc_generics.models import (
+    OpenIDConnectDigiDConfig,
+    OpenIDConnectEHerkenningConfig,
+)
+from open_inwoner.configurations.models import SiteConfiguration
+
+
+class DigiDOIDCConfigurationStep(BaseConfigurationStep):
+    """
+    Configure DigiD authentication via OpenID Connect
+    """
+
+    verbose_name = "Configuration for DigiD via OpenID Connect"
+    required_settings = [
+        "DIGID_OIDC_OIDC_RP_CLIENT_ID",
+        "DIGID_OIDC_OIDC_RP_CLIENT_SECRET",
+        # NOTE these are part of the model, but not actually part of the admin form
+        # "DIGID_OIDC_OIDC_USE_NONCE",
+        # "DIGID_OIDC_OIDC_NONCE_SIZE",
+        # "DIGID_OIDC_OIDC_STATE_SIZE",
+        # "DIGID_OIDC_OIDC_EXEMPT_URLS",
+    ]
+    all_settings = required_settings + [
+        "DIGID_OIDC_IDENTIFIER_CLAIM_NAME",
+        "DIGID_OIDC_OIDC_RP_SCOPES_LIST",
+        "DIGID_OIDC_OIDC_RP_SIGN_ALGO",
+        "DIGID_OIDC_OIDC_RP_IDP_SIGN_KEY",
+        "DIGID_OIDC_OIDC_OP_DISCOVERY_ENDPOINT",
+        "DIGID_OIDC_OIDC_OP_JWKS_ENDPOINT",
+        "DIGID_OIDC_OIDC_OP_AUTHORIZATION_ENDPOINT",
+        "DIGID_OIDC_OIDC_OP_TOKEN_ENDPOINT",
+        "DIGID_OIDC_OIDC_OP_USER_ENDPOINT",
+        "DIGID_OIDC_OIDC_OP_LOGOUT_ENDPOINT",
+        "DIGID_OIDC_USERINFO_CLAIMS_SOURCE",
+        "DIGID_OIDC_ERROR_MESSAGE_MAPPING",
+        "DIGID_OIDC_OIDC_KEYCLOAK_IDP_HINT",
+    ]
+    enable_setting = "DIGID_OIDC_ENABLE"
+
+    def is_configured(self) -> bool:
+        return OpenIDConnectDigiDConfig.get_solo().enabled
+
+    def configure(self):
+        config = OpenIDConnectDigiDConfig.get_solo()
+
+        # Use the model defaults
+        form_data = {
+            field.name: getattr(config, field.name)
+            for field in OpenIDConnectDigiDConfig._meta.fields
+        }
+
+        # Only override field values with settings if they are defined
+        for setting in self.all_settings:
+            value = getattr(settings, setting, None)
+            if value is not None:
+                model_field_name = setting.split("DIGID_OIDC_")[1].lower()
```

Comment:
I rely on the name being prefixed by `DIGID_OIDC_` and the rest of the variable name being identical to the name of the model field. Some fields on the config start with `oidc_`, that's why some settings contain `OIDC_` twice (e.g. `DIGID_OIDC_OIDC_RP_CLIENT_ID`).